### PR TITLE
Feature/gh 41

### PR DESCRIPTION
**Pull request by rwi**
- cleanup of mailing API:
  - MailSender (non-transactional Spring service) instead of MailManagement
  - API for testing new mail settings which is used by installer and admin configuration controller
  - MailMessage no longer implements Spring's MimeMessagePreparator
- factory which creates mail sender implementation based on development properties
  - renamed the properties for enabling and configuring the catch-all message sender. Original recipients are stored in special mail headers now instead of adding them to the mail body.
- define development.properties in automatic integration tests so that factory creates FileSystemMimeMessageSender
- some renaming (esp. receiver renamed to recipient) and code documentation

### CODE REVIEW

File: communote/api/src/main/java/com/communote/server/api/core/config/DevelopmentProperties.java [hunk 1]
```diff
@@ -59,37 +59,17 @@ public String getMailingTestAddress() {
      * @param configPath
      *            path to the configuration directory
      */
-    private void initPropertiesFromFile(File configPath) {
+    private Properties initPropertiesFromFile(File configPath) {
         File propsFile = new File(configPath, DEVEL_PROPERTIES_FILENAME);
         if (!propsFile.exists()) {
-            // return silently
-            return;
+            // ignore silently
```
Comment (scrobbleme) [inline communote/api/src/main/java/com/communote/server/api/core/config/DevelopmentProperties.java]: @rwi Evtl. ein Log auf DEBUG. Im Zweifel lieber explizit statt implizit (durch das Log dadrunter)

File: communote/api/src/main/java/com/communote/server/api/core/config/DevelopmentProperties.java [hunk 2]
```diff
@@ -98,6 +78,90 @@ private void initPropertiesFromFile(File configPath) {
...
+     * @param fallback
+     *            the fallback to return if there is no property for the key
+     * @return the property value or the fallback if there is no property for the key or it cannot
+     *         be converted into an integer
+     */
+    public int getProperty(String key, int fallback) {
+        try {
+            return Integer.parseInt(getProperty(key));
+        } catch (NumberFormatException e) {
+            return fallback;
```
Comment (scrobbleme) [inline communote/api/src/main/java/com/communote/server/api/core/config/DevelopmentProperties.java]: @rwi Evtl. Log auf DEBUG?

File: communote/api/src/main/java/com/communote/server/api/core/config/DevelopmentProperties.java [hunk 3]
```diff
@@ -98,6 +78,90 @@ private void initPropertiesFromFile(File configPath) {
...
+     * @param fallback
+     *            the fallback to return if there is no property for the key
+     * @return the property value or the fallback if there is no property for the key or it cannot
+     *         be converted into a long
+     */
+    public long getProperty(String key, long fallback) {
+        try {
+            return Long.parseLong(getProperty(key));
+        } catch (NumberFormatException e) {
+            return fallback;
```
Comment (scrobbleme) [inline communote/api/src/main/java/com/communote/server/api/core/config/DevelopmentProperties.java]: @rwi Evtl. Log auf DEBUG?

File: communote/core/src/main/java/com/communote/server/core/blog/MailBasedPostingManagementImpl.java [hunk 4]
```diff
@@ -576,7 +576,7 @@ private User resolveUser(String senderEmail) {
      *            the error mail message
      */
     private void sendErrorMailMessage(MailMessage message) {
-        ServiceLocator.instance().getService(MailManagement.class).sendMail(message);
+        ServiceLocator.instance().getService(MailSender.class).send(message);
```
Comment (scrobbleme) [inline communote/core/src/main/java/com/communote/server/core/blog/MailBasedPostingManagementImpl.java]: @rwi Kann man den Sender hier nicht per DI reinbekommen?

File: communote/core/src/main/java/com/communote/server/core/mail/MailMessageHelper.java [hunk 5]
```diff
@@ -79,105 +76,29 @@
...
-     *            The addresses to change, if null nothing will be done
-     * @param newAddress
-     *            The new one
-     */
-    public static void changeAddresses(StringBuilder output, Address[] addresses, String newAddress) {
-        if (addresses != null) {
-            for (Address address : addresses) {
-                changeAddress(output, address, newAddress);
+        for (Address a : addresses) {
+            if (a instanceof InternetAddress) {
```
Comment (scrobbleme) [inline communote/core/src/main/java/com/communote/server/core/mail/MailMessageHelper.java]: @rwi Was wäre denn die Alternatve? Evtl. Warnung loggen?

**Comment by rwi** [comment]
@scrobbleme danke für dein Code Review!

**Pull request merged**